cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/OverlayNode.kt

```diff
@@ -54,9 +54,21 @@ abstract class OverlayNode() : Node() {
      * have different underlying nodes.
      */
     override fun equals(other: Any?): Boolean {
+        if (this === other) {
+            return true
+        }
         return other is OverlayNode &&
-            super.equals(other) &&
-            this.underlyingNode == other.underlyingNode
+            name == other.name &&
+            code == other.code &&
+            comment == other.comment &&
+            location == other.location &&
+            isImplicit == other.isImplicit &&
+            // If the underlying node is null, we don't care about it. We also cannot compare the
+            // location in this case.
+            (this.underlyingNode == null ||
```

Comment:
Hmm I am pretty sure I had proposed similar code in the past and I was widely criticized for that :D I thought we don't need this special handling of an empty underlying node.

Maybe I misunderstood what the original problem is?